doc/yjit/yjit_hacking.md

```diff
@@ -0,0 +1,79 @@
+# YJIT Hacking
+
+## Code Generation and Assembly Language
+
+YJIT’s basic purpose is to take ISEQs and generate machine code.
+
+Documentation on each Ruby bytecode can be found in insns.def.
+
+YJIT uses those bytecodes as the “Basic Blocks” in Lazy Basic Block Versioning (LBBV.) For more deep details of LBBV:
+
+* Basic Block Versioning (whitepaper) - https://arxiv.org/abs/1411.0352
+* An Extension of BBV using Object Shapes (whitepaper) - https://arxiv.org/abs/1507.02437
+* Basic Block Versioning talk at ECOOP 2015 - https://www.youtube.com/watch?v=S-aHBuoiYE0
```

Comment:
We have links to both of these papers and accompanying talks in the current README.md